mono/utils/mono-safe-handle.h

```diff
@@ -0,0 +1,42 @@
+/**
+ * \file
+ * Support for synchronously freeing unmanaged memory subject to
+ * concurrent refcount modification
+ *
+ * Author:
+ *   Alexander Kyte (alkyte@microsoft.com)
+ *
+ * (C) 2018 Microsoft, Inc.
+ *
+ */
+#ifndef __MONO_UTILS_SAFE_HANDLE__
+#define __MONO_UTILS_SAFE_HANDLE__
+
+#include "mono/utils/mono-coop-mutex.h"
+
+typedef union {
+	gint64 mem_start;
```

Comment:
I'd call it i or integer or as_int or asint.
AsInt32 and such is the pattern in other projects

Why did we go with gint64 vs. gsize? Want same size across BCL builds? Or just easier to reason about fixed size? Or worried about 32bit overflow on 32bit system?

Reply:
In particular 64bit CAS operations are cheap on modern 32bit x86, but what about 32bit ARM?